micrometer-tracing-bridges/micrometer-tracing-bridge-brave/src/main/java/io/micrometer/tracing/brave/bridge/BraveBaggageManager.java

```diff
@@ -36,15 +37,25 @@ public class BraveBaggageManager implements Closeable, BaggageManager {
 
     private final List<String> tagFields;
 
+    private final List<String> remoteFields;
+
     @Nullable
     private Tracer tracer;
 
     /**
      * Create an instance of {@link BraveBaggageManager}.
      * @param tagFields fields of baggage keys that should become tags on a span
+     * @param remoteFields fields of baggage keys that should be propagated over the wire
      */
-    public BraveBaggageManager(List<String> tagFields) {
+    public BraveBaggageManager(List<String> tagFields, List<String> remoteFields) {
         this.tagFields = tagFields;
+        this.remoteFields = remoteFields(tagFields, remoteFields);
+    }
+
+    private static List<String> remoteFields(List<String> tagFields, List<String> remoteFields) {
+        List<String> combined = new ArrayList<>(tagFields);
+        combined.addAll(remoteFields);
```

Comment:
Why are these combined? It looks like the other baggage managers don't combine them.

Reply:
Is it better to create a unique set of fields?
The values seem to be used in the handler for `contains`, so in the end, it doesn't need to be unique here, but just for clarity.

Reply:
I think I'm still confused. The assumption seems to be that all tagFields are remote baggage. Is local baggage that should also be tagged not possible? In the test code, it looks like the baggage key to be tagged is local baggage:

```
.add(BaggagePropagationConfig.SingleBaggageField.local(BaggageField.create(TAG_KEY)))
```

So shouldn't that not be combined with remote fields?

Reply:
From the point of view of a baggage manager there is no concept of local fields. That is really brave specific and has to be configured via propagation configuration. I understand that this might be confusing. We can rename this from `remoteFields` to `baggageFields` and that way that will mean all baggage regardless of whether it's local or remote. WDYT?

Reply:
That sounds good.